File: JSPM.Web/Modules/ServerTypes/Organisation/DepartmentsRow.ts
import { fieldsProxy } from "@serenity-is/corelib/q";

export interface DepartmentsRow {
    Id?: number;
    DepartmentName?: string;
}

export abstract class DepartmentsRow {
    static readonly idProperty = 'Id';
    static readonly nameProperty = 'DepartmentName';
    static readonly localTextPrefix = 'Organisation.Departments';
    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<DepartmentsRow>();
}

File: JSPM.Web/Modules/Organisation/Departments/DepartmentsRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace JSPM.Organisation;

[ConnectionKey("Default"), Module("Organisation"), TableName("[organisation].[Departments]")]
[DisplayName("Departments"), InstanceName("Departments")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
public sealed class DepartmentsRow : Row<DepartmentsRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Id"), Identity, IdProperty]
    public int? Id
    {
        get => fields.Id[this];
        set => fields.Id[this] = value;
    }

    [DisplayName("Department Name"), Column("Department Name"), Size(200), NotNull, QuickSearch, NameProperty]
    public string DepartmentName
    {
        get => fields.DepartmentName[this];
        set => fields.DepartmentName[this] = value;
    }

    public class RowFields : RowFieldsBase
    {
        public Int32Field Id;
        public StringField DepartmentName;

    }
}